pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -455,7 +455,8 @@ def _share_information_to_prevent_deadlock(self) -> None:
     def _share_pids(self) -> None:
         """Make all DDP processes aware of all processes pids."""
         self.barrier()
-        pids = self.all_gather(torch.tensor(os.getpid(), device=self.root_device))
```

Comment:
Let's rely on self.all_gather instead.

Reply:
May I ask what `rely on` mean? You mean just using self.all_gather instead of pytorch's all_gather?

Reply:
Yes